// <auto-generated>
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for
// license information.
//
// Code generated by Microsoft (R) AutoRest Code Generator.
// Changes may cause incorrect behavior and will be lost if the code is
// regenerated.
// </auto-generated>

namespace Microsoft.Azure.CognitiveServices.Vision.ComputerVision.Models
{
    using Microsoft.Rest;

    /// <summary>
    /// Exception thrown for an invalid response with ComputerVisionOcrError
    /// information.
    /// </summary>
    public partial class ComputerVisionOcrErrorException : RestException
    {
        /// <summary>
        /// Gets information about the associated HTTP request.
        /// </summary>
        public HttpRequestMessageWrapper Request { get; set; }

        /// <summary>
        /// Gets information about the associated HTTP response.
        /// </summary>
        public HttpResponseMessageWrapper Response { get; set; }

        /// <summary>
        /// Gets or sets the body object.
        /// </summary>
        public ComputerVisionOcrError Body { get; set; }

        /// <summary>
        /// Initializes a new instance of the ComputerVisionOcrErrorException class.
        /// </summary>
        public ComputerVisionOcrErrorException()
        {
        }

        /// <summary>
        /// Initializes a new instance of the ComputerVisionOcrErrorException class.
        /// </summary>
        /// <param name="message">The exception message.</param>
        public ComputerVisionOcrErrorException(string message)
            : this(message, null)
        {
        }

        /// <summary>
        /// Initializes a new instance of the ComputerVisionOcrErrorException class.
        /// </summary>
        /// <param name="message">The exception message.</param>
        /// <param name="innerException">Inner exception.</param>
        public ComputerVisionOcrErrorException(string message, System.Exception innerException)
            : base(message, innerException)
        {
        }
    }
}
